### DISCUSSION TITLE
Android and remote control

### DISCUSSION THREAD

**Discussion by decodering22** [GerryDazoo/Slinger#59] [Q&A] [answered]
Thanks for making this possible. For those of us who used the slingbox app on our phones. I assume the same applies by running VLC app on android correct with setting the correct address and port? 

I have not tested this yet but it seems having to jump between a browser and VLC on a phone it would be difficult to know if your changes are taking effect without jumping back and forth. I'm not sure if it is possible to use this on mobile as intended. 

Lastly with this server running, is there a way to integrate the ability to use the existing sling app on mobile, likely not without major reverse engineering the app smali etc. but can't hurt to ask?

**Comment by GerryDazoo** [comment, marked as answer]
Many questions.
Yes, just point VLC on ypor phone to http://yourserver:port/slingbox. I do this all the time. I have had some issues with audio sync on the VLC android app. I've been using MX player.

Depending on your phone you can do  split screen or put the vlc app into background while using the browser. Not ideal but useable for changing channels and simple navigation.

You can use both but not at the same time. The slingbox can only stream to a single server at one time. If you startup a session with my code it will kick off any existing users. But it's fine to toggle back and forth is you really want to.

**Reply by decodering22** [reply]
Thanks Gerry, i don't think my phone support split screen. And while this is not related to your brilliant work around I'm curious if you have a Tivo yourself and got a slingbox like a lot of us as a work around b/c of their app and inability to stream a lot of shows marked at copyright to prevent streaming away from the home. 

If you have Tivo have you looked at possibly getting around the copyright flag for being able to stream recorded shows away from the home? I've tried VPN with no luck. Quite frustrating.

**Reply by GerryDazoo** [reply]
Once the video is encoded by the slingbox there is no copyright protection, If your Tivo can play it the slingbox/my code can stream it. I don't have a Tivo but many of my users do.
To stream away form home you can in your config.ini add your home router external IPaddress and in your home router add a port forwarding rule to map a port to your home slingbox.

**Reply by decodering22** [reply]
The issue isn't with slingbox, i can use the slingbox now to view copyright protected flagged apps by Tivo. What i'm hoping is to get a Tivo app alternative to a way to fool the existing Tivo app that would allow you to view any recorded show by either tricking the Tivo app into thinking you are at your home network or an alternative app that replaces the tivo app. The latter is not likely but if there is a way to trick the existing Tivo app via VPN or other means it would make the slingbox moot for those of us who have a Tivo and resorted to a slingbox in the first place to stream those copyright flagged shows.

**Reply by tom76017** [reply]
VLC works fine for me on any device both internal and external.  I haven't figured out how change the buffer time.  For the remote, I'm using the Peanut TiVo app on my iPhone with a TiVo Mini.  Take a look at this https://github.com/GerryDazoo/Slinger/discussions/30

**Reply by GerryDazoo** [reply]
That's good news. The next version will handle your configuration so you don't have to make any code changes.
